tests/e2e/multi_svc_test.go

```diff
@@ -0,0 +1,658 @@
+			var roleUser0, roleUser1 string
+
+			switch {
+			case strings.Contains(datastore, "local"):
+				roleUser0 = roleCnsSearchAndSpbm
+				roleUser1 = roleCnsSearchAndSpbm
+			case strings.Contains(datastore, "nfs"):
+				roleUser0 = roleCnsDatastore
+				roleUser1 = roleCnsDatastore
+			case strings.Contains(datastore, "vsanDatastore (1)"):
+				roleUser0 = roleCnsSearchAndSpbm
+				roleUser1 = roleCnsDatastore
+			default:
+				roleUser0 = roleCnsDatastore
+				roleUser1 = roleCnsSearchAndSpbm
+			}
+
+			userPermission, err = verifyPermissionForWcpStorageUser(dataCenter, datastore, "", "", wcpServiceAccUsers[0], roleUser0)
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+			gomega.Expect(userPermission).To(gomega.BeTrue(), "user permission is not valid for datastore")
+			userPermission, err = verifyPermissionForWcpStorageUser(dataCenter, datastore, "", "", wcpServiceAccUsers[1], roleUser1)
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+			gomega.Expect(userPermission).To(gomega.BeTrue(), "user permission is not valid for datastore")
+		}
+
+	})
+
+	/* TESTCASE-4
+
+	Verify that alarm an alarm is raised when a shared datastore becomes non-shared
+
+	Steps:
+		1. add a nfs shared datastore
+		2. verify that required permissions are set for the datastore
+		3. unmount nfs datastore from one of the hosts say host1 and verify that an alarm is raised indicating that a shared datastore is no longer shared now
+		4. remove the host1 from the cluster
+		5. verify that the alarm from step 4 is cleared
+		6. add host1 back to cluster and don't mount nfs datastore on it yet and verify that an alarm is raised again indicating that a shared datastore is no longer shared across all hosts in the cluster now
+		7. mount nfs datastore on host1 and verify that the alarm from step 7 is gone
+	*/
+
+	ginkgo.It("[csi-multi-svc] Verify that alarm an alarm is raised when a shared datastore becomes non-shared", ginkgo.Label(p0, multiSvc, stable), func() {
+		ctx, cancel := context.WithCancel(context.Background())
+		defer cancel()
+
+		var timeout int32 = 300
+		unMount := false
+		hostInMM := false
+
+		ginkgo.By("Adding a shared datastore to supervisor cluster")
+		err := mountDatastoreOnClusterOrHost(datastoreName, datastoreIP, computeClusterPaths[0])
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		defer func() {
+			ginkgo.By("Remove mounted datastore from cluster")
+			UnMountDatastoreFromClusterOrHost(datastoreName, computeClusterPaths[0])
+		}()
+
+		ginkgo.By("Verify datastore has permission for storage service account from supervisor cluster")
+		userPermission, err := verifyPermissionForWcpStorageUser(dataCenter, datastoreName, "", "", wcpServiceAccUsers[0], roleCnsDatastore)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		gomega.Expect(userPermission).To(gomega.BeTrue(), "user permission is not changed for cluster")
+
+		ginkgo.By("Unmount datastore from one of the host from supervisor cluster")
```

Comment:
put it in a util and make util in a more generic way

Reply:
I will make it more generic as per the needs, for now looks good